### PR TITLE
Templated benchmarks improvements

### PR DESCRIPTION
**Pull request by bpkroth**
- enable testing locally via docker build-run script
- enable random seed from config
- enable sqlite support
- use sqlite support to implement unit tests
- use unit tests to walk through debugger to find error (csv parsing)
- [ ] left some TODO comments about adding support for NULL processing
  - I handled one of these, but there's some lingering work that should probably be taken up separately for improving this to be able to pass `NULL`able values as a parameter.
    For instance, `c VARCHAR(255) NULL` type columns can't receive a NULL value atm via something like `<value />` since the values are passed to the constructors as a comma separated string list, which gets parsed back out as an empty string.

### CODE REVIEW

File: data/templated/example.xml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Note: these example templates reuse TPC-C queries and data as a demonstration, but others are possible. -->
+<!-- TODO: Add additional parameter data types to increase test coverage. -->
```
Comment (bpkroth) [inline data/templated/example.xml:3]: Especially NULLs

File: src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java [hunk 2]
```diff
@@ -63,11 +63,16 @@ public PreparedStatement getStatement(Connection conn, List<Object> params) thro
         String[] paramsTypes = queryTemplateInfo.getParamsTypes();
         for (int i = 0; i < paramsTypes.length; i++) {
             if (paramsTypes[i].equalsIgnoreCase("NULL")) {
-                stmt.setNull(i + 1, Types.NULL);
+                stmt.setNull(i + 1, Types.NULL);    // TODO: Add test case for this
```
Comment (bpkroth) [inline src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java]: TODO

File: src/test/java/com/oltpbenchmark/api/AbstractTestCase.java [hunk 3]
```diff
@@ -138,6 +133,8 @@ public final void setUp() throws Exception {
                 new Class<?>[]{WorkloadConfiguration.class});
         assertNotNull(this.benchmark);
 
+        assertNotNull(this.procedures());
```
Comment (bpkroth) [inline src/test/java/com/oltpbenchmark/api/AbstractTestCase.java:140]: Calling a second time to make the templated version work correctly.

File: src/test/java/com/oltpbenchmark/api/AbstractTestCase.java [hunk 4]
```diff
@@ -174,6 +159,38 @@ public final void setUp() throws Exception {
...
+        int id = 0;
+        for (Class<? extends Procedure> procedureClass : procedures) {
+            TransactionType tt = new TransactionType(procedureClass, id++, false, 0, 0);
+            txnTypes.add(tt);
+        }
+
+        return txnTypes;
+    }
+
+    protected void createDatabase() {
```
Comment (bpkroth) [inline src/test/java/com/oltpbenchmark/api/AbstractTestCase.java:178]: Allows overriding in templated benchmark.

Comment (anjagruenheid) [reply src/test/java/com/oltpbenchmark/api/AbstractTestCase.java:178]: Makes sense!

File: data/templated/example.xml (hunk 1 above)
Comment (bpkroth) [reply data/templated/example.xml:3]: Did this for explicit NULL types, but not "NULLable" types (e.g., `c VARCHAR NULL` type columns).
The issue here is that we convert params to CSVs lists when passing them to objects which changes those values into "empty" strings instead.

File: src/test/java/com/oltpbenchmark/api/AbstractTestCase.java (hunk 3 above)
Comment (bpkroth) [reply src/test/java/com/oltpbenchmark/api/AbstractTestCase.java:140]: Commented.

File: src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java (hunk 2 above)
Comment (bpkroth) [reply src/main/java/com/oltpbenchmark/benchmarks/templated/procedures/GenericQuery.java]: done

File: data/templated/example.xml (hunk 1 above)
Comment (bpkroth) [reply data/templated/example.xml:3]: Let's take this up in future work.

**Pull request merged**